scout/server/blueprints/variant/utils.py

```diff
@@ -0,0 +1,453 @@
+
+        if not hgnc_gene:
+            continue
+        hgnc_symbol = hgnc_gene['hgnc_symbol']
+        # Add omim information if gene is annotated to have incomplete penetrance
+        if hgnc_gene.get('incomplete_penetrance'):
+            variant_gene['omim_penetrance'] = True
+
+        ############# PANEL SPECIFIC INFORMATION #############
+        # Panels can have extra information about genes and transcripts
+        panel_info = add_panel_specific_gene_info(extra_info.get(hgnc_id, []), variant_gene)
+        variant_gene.update(panel_info)
+        
+        update_transcripts_information(variant_gene, hgnc_gene, variant_obj)
+        
+        variant_gene['common'] = hgnc_gene
+        
+        add_gene_links(variant_gene, genome_build)
+        
+        # Add disease associated transcripts to variant
+        for refseq_id in panel_info.get('disease_associated_transcripts', []):
+            transcript_str = "{}:{}".format(hgnc_symbol, refseq_id)
+            variant_obj['disease_associated_transcripts'].append(transcript_str)
+        
+        # Add the associated disease terms
+        disease_terms = store.disease_terms(hgnc_id)
+        variant_gene['disease_terms'] = disease_terms
+
+        all_models = all_models.union(set(variant_gene['manual_inheritance']))
+        omim_models = set()
+        for disease_term in variant_gene.get('disease_terms', []):
+            omim_models.update(disease_term.get('inheritance', []))
+        variant_gene['omim_inheritance'] = list(omim_models)
+
+        all_models = all_models.union(omim_models)
+    
+    variant_obj['all_models'] = all_models
+
+    return variant_obj
+
+def predictions(genes):
+    """Adds information from variant specific genes to display.
+    
+    Args:
+        genes(list[dict])
+    
+    Returns:
+        data(dict)
+    """
+    data = {
+        'sift_predictions': [],
+        'polyphen_predictions': [],
+        'region_annotations': [],
+        'functional_annotations': []
+    }
+    for gene_obj in genes:
+        for pred_key in data:
+            gene_key = pred_key[:-1]
+            if len(genes) == 1:
+                value = gene_obj.get(gene_key, '-')
+            else:
+                gene_id = gene_obj.get('hgnc_symbol') or str(gene_obj.get('hgnc_id',0))
+                value = ':'.join([gene_id, gene_obj.get(gene_key, '-')])
+            data[pred_key].append(value)
+
+    return data
+
+def sv_frequencies(variant_obj):
+    """Add frequencies in the correct way for the template
+    
+    This function converts the raw annotations to something better to visualize
+    
+    Args:
+        variant_obj(scout.models.Variant)
+    
+    Returns:
+        frequencies(list(tuple)): A list of frequencies to display
+    """
+    
+    # Mandatory frequencies
+    
+    frequencies = [
+        ('GnomAD', variant_obj.get('gnomad_frequency')),
+    ]
+
+    if 'clingen_cgh_benign' in variant_obj:
```

Comment:
A dict and a loop instead perhaps?

Reply:
Fixed that, good idea. This will make it easy to expand with more frequencies